packages/network-controller/src/index.ts

```diff
@@ -1,6 +1,11 @@
+export type { AutoManagedNetworkClient } from './create-auto-managed-network-client';
 export * from './NetworkController';
 export * from './constants';
 export type { BlockTracker, Provider } from './types';
-export type { NetworkClientConfiguration } from './types';
+export type {
+  NetworkClientConfiguration,
+  InfuraNetworkClientConfiguration,
+  CustomNetworkClientConfiguration,
+} from './types';
```

Comment:
These exports accommodate the new import statements in `Token{Detection,Rate}Controller.test.ts`.

Previously, these types were accessed through subpath imports, which are now not supported unless specified in the library's package.json `exports` field.